### ISSUE TITLE
`tilt ci` doesn't terminate once all containers are ready

### ISSUE THREAD

**Issue by greglanthier** [tilt-dev/tilt#4133]
This feels like a variant of issue #3262 in the sense that `tilt ci` appears to hang indefinitely after a workload has been deployed to a Kubernetes cluster.

### Expected Behavior

`tilt ci` will exit once the `kubebuilder` operator from the reproduce below is deployed and ready in a local dev Kubernetes cluster.

### Current Behavior

`tilt ci` doesn't exit once all of the containers in the operator pod are ready.

### Steps to Reproduce
1. Provision a local kubernetes cluster (`kind`, `minikube`, `k3s`, it shouldn't matter).  Make sure your `~/.kube/config` points at that cluster.
2. Run the following in an empty directory:
```bash
cat <<EOF > Tiltfile
load('ext://kubebuilder', 'kubebuilder')
kubebuilder("domain.tilt.dev", "tiltgroup", "v1", "CustomKind")
EOF
tilt ci
```
3. Observe that the `tilt ci` command doesn't terminate even after all of the containers in the `controller-manager` pod are ready.

#### `tilt doctor` Output

```bash
$ tilt doctor

Tilt: v0.18.5, built 2021-01-22
System: darwin-amd64
---
Docker
- Host: [default]
- Version: 1.41
- Builder: 2
---
Kubernetes
- Env: kind-0.6+
- Context: kind-kind
- Cluster Name: kind-kind
- Namespace: default
- Container Runtime: containerd
- Version: v1.20.2
- Cluster Local Registry: none
---
Thanks for seeing the Tilt Doctor!
Please send the info above when filing bug reports. 💗

The info below helps us understand how you're using Tilt so we can improve,
but is not required to ask for help.
---
Analytics Settings
--> (These results reflect your personal opt in/out status and may be overridden by an `analytics_settings` call in your Tiltfile)
- User Mode: opt-out
- Machine: df1a2bcef66ccf958d6affd19a2cd1c9
- Repo: 
```

#### About Your Use Case

The `tilt ci` command provides an ideal way of performing smoke tests on Kubernetes deployment specs & container images.  We have CI builds that worked using the `tilt ci` incantation with `tilt 0.18.3` but newer builds of `tilt` won't work on our local machines.

This issue prevents developers from using the `tilt ci` command locally to replicate the steps running in our CI builds.

The attached [`tilt-dump.json.txt`](https://github.com/tilt-dev/tilt/files/5883134/tilt-dump.json.txt) was generated from the lingering `tilt ci` incantation after deploying the operator to a local kind cluster running on my laptop.

**Comment by landism** [comment]
Thanks for the straightforward repro! I'll take a look.

**Comment by landism** [comment]
This was broken by #4082 in [Tilt 0.18.5](https://github.com/tilt-dev/tilt/releases/tag/v0.18.5). `tilt ci` no longer correctly determines `local_resource`s without `serve_cmd`s are done. We should have a fix or revert in the next day or two.

**Comment by greglanthier** [comment]
Wow, thanks for the super fast fix, @milas!  I'll give this a try as soon as there is a release.

**Comment by milas** [comment]
We just released [v0.18.6](https://github.com/tilt-dev/tilt/releases/latest) which includes this fix. Thanks again for making an issue!
